libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -613,32 +650,51 @@ void  NavEKF3_core::updateFilterStatus(void)
     filterStatus.flags.takeoff = expectTakeoff; // The EKF has been told to expect takeoff is in a ground effect mitigation mode and has started the EKF-GSF yaw estimator
     filterStatus.flags.touchdown = expectGndEffectTouchdown; // The EKF has been told to detect touchdown and is in a ground effect mitigation mode
     filterStatus.flags.using_gps = ((imuSampleTime_ms - lastPosPassTime_ms) < 4000) && (PV_AidingMode == AID_ABSOLUTE);
-    filterStatus.flags.gps_glitching = !gpsAccuracyGood && (PV_AidingMode == AID_ABSOLUTE) && (frontend->_fusionModeGPS != 3); // GPS glitching is affecting navigation accuracy
+    filterStatus.flags.gps_glitching = !gpsAccuracyGood && (PV_AidingMode == AID_ABSOLUTE) && (frontend->sources.getPosXYSource() == AP_NavEKF_Source::SourceXY::GPS); // GPS glitching is affecting navigation accuracy
     filterStatus.flags.gps_quality_good = gpsGoodToAlign;
     filterStatus.flags.initalized = filterStatus.flags.initalized || healthy();
 }
 
 void NavEKF3_core::runYawEstimatorPrediction()
 {
-    if (yawEstimator != nullptr && frontend->_fusionModeGPS <= 1) {
-        float trueAirspeed;
-        if (is_positive(defaultAirSpeed) && assume_zero_sideslip()) {
-            if (imuDataDelayed.time_ms - tasDataDelayed.time_ms < 5000) {
-                trueAirspeed = tasDataDelayed.tas;
-            } else {
-                trueAirspeed = defaultAirSpeed * dal.get_EAS2TAS();
-            }
+    // exit immediately if no yaw estimator
+    if (yawEstimator == nullptr) {
+        return;
+    }
+
+    // ensure GPS is used for horizontal position and velocity
+    if (frontend->sources.getPosXYSource() != AP_NavEKF_Source::SourceXY::GPS ||
+        !frontend->sources.useVelXYSource(AP_NavEKF_Source::SourceXY::GPS)) {
+        return;
+    }
+
+    float trueAirspeed;
+    if (is_positive(defaultAirSpeed) && assume_zero_sideslip()) {
+        if (imuDataDelayed.time_ms - tasDataDelayed.time_ms < 5000) {
+            trueAirspeed = tasDataDelayed.tas;
         } else {
-            trueAirspeed = 0.0f;
+            trueAirspeed = defaultAirSpeed * dal.get_EAS2TAS();
         }
-
-        yawEstimator->update(imuDataDelayed.delAng, imuDataDelayed.delVel, imuDataDelayed.delAngDT, imuDataDelayed.delVelDT, EKFGSF_run_filterbank, trueAirspeed);
+    } else {
+        trueAirspeed = 0.0f;
     }
+
+    yawEstimator->update(imuDataDelayed.delAng, imuDataDelayed.delVel, imuDataDelayed.delAngDT, imuDataDelayed.delVelDT, EKFGSF_run_filterbank, trueAirspeed);
 }
 
 void NavEKF3_core::runYawEstimatorCorrection()
 {
-    if (yawEstimator != nullptr && frontend->_fusionModeGPS <= 1 && EKFGSF_run_filterbank) {
+    // exit immediately if no yaw estimator
+    if (yawEstimator == nullptr) {
+        return;
+    }
+    // ensure GPS is used for horizontal position and velocity
```

Comment:
yes, I agree.  It could be very useful if the GSF could work off of velocities provided by an external nav system.  Still, it's out of scope at this point.  I suspect their could be issues around ensuring the velocities are accurate.